test/circuit.jl

```diff
@@ -38,7 +50,34 @@ end
 
     append!(c, [sigma_x(2), sigma_y(2)])
 
-    @test_throws DomainError push!(c, sigma_x(5))
+    @test_throws DomainError(
+        5,
+        "The instruction does not fit in the circuit: " * "target qubit: 5, qubit_count: 3",
```

Comment:
Similar to before, checking the exact error string can become a pain later on... can we change this to check just the type (and we'll introduce/use a particular type for testing later if need be).

Reply:
Oh, nvm... we can land this change as-is and fix these incrementally.